pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -414,3 +434,124 @@ func (c *EtcdProxyController) handleObject(obj interface{}) {
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceReclaimed, msg)
+			return err
+		}
+		msg := fmt.Sprintf(ResourceReclaimedReason, caConfigMap.Name)
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ResourceReclaimed, msg)
+	}
+
+	return nil
+}
+
+// createEtcdProxyCerts creates Secret in controller namespace with Etcd Proxy Server certificate and key.
+func (c EtcdProxyController) createEtcdProxyCerts(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
+	// Secret for the etcd proxy server certs in controller namespace.
+	certsSecret, err := c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+		Get(etcdProxyServerCertsSecret(etcdstorage), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		data := make(map[string]string, 2)
+		data["server.pem"] = EtcdProxyServerCert
+		data["server-key.pem"] = EtcdProxyServerKey
+		certsSecret, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+			Create(newSecret(etcdstorage, etcdProxyServerCertsSecret(etcdstorage), c.config.ControllerNamespace, data))
+	}
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+	if !metav1.IsControlledBy(certsSecret, etcdstorage) {
+		certsSecret.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
+		})
+		certsSecret, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).Update(certsSecret)
+		if err != nil {
+			msg := fmt.Sprintf(MessageErrResourceReclaimed, certsSecret.Name)
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceReclaimed, msg)
+			return err
+		}
+		msg := fmt.Sprintf(ResourceReclaimedReason, certsSecret.Name)
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ResourceReclaimed, msg)
+	}
+
+	return nil
+}
+
+// updateEtcdProxyAPIServerCACert updates the ConfigMap in the aggregated API server namespace with the CA certificate.
+func (c *EtcdProxyController) updateEtcdProxyAPIServerCACert(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
+	// Check are ConfigMap name and namespace provided.
+	if len(etcdstorage.Spec.CACertConfigMap) == 0 {
+		err := fmt.Errorf("CA ConfigMap name and namespace not provided")
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+
+	for _, configMap := range etcdstorage.Spec.CACertConfigMap {
+		caConfigMap, err := c.kubeclientset.CoreV1().ConfigMaps(configMap.Namespace).
+			Get(configMap.Name, metav1.GetOptions{})
+		if err != nil {
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
+		}
+		if len(caConfigMap.Data) < 1 {
+			caConfigMap.Data = make(map[string]string, 1)
+		}
+		caConfigMap.Data["ca.pem"] = EtcdProxyCACert
+		_, err = c.kubeclientset.CoreV1().ConfigMaps(caConfigMap.Namespace).Update(caConfigMap)
+
+		return err
```

Comment:
do you have tests that on error (e.g. conflicts) the controller keep retrying?

Reply:
This one is still not solved.